container.go

```diff
@@ -542,6 +542,128 @@ func (c *Client) TopContainer(id string, psArgs string) (TopResult, error) {
+			//HierarchicalMemoryLimit int64 `json:"hierarchical_memory_limit,omitempty" yaml:"hierarchical_memory_limit,omitempty"`
+			TotalPgfault    int64 `json:"total_pgfault,omitempty" yaml:"total_pgfault,omitempty"`
+			TotalActivefile int64 `json:"total_active_file,omitempty" yaml:"total_active_file,omitempty"`
+			ActiveAnon      int64 `json:"active_anon,omitempty" yaml:"active_anon,omitempty"`
+			TotalActiveAnon int64 `json:"total_active_anon,omitempty" yaml:"total_active_anon,omitempty"`
+			TotalPgpgout    int64 `json:"total_pgpgout,omitempty" yaml:"total_pgpgout,omitempty"`
+			TotalCache      int64 `json:"total_cache,omitempty" yaml:"total_cache,omitempty"`
+			InactiveAnon    int64 `json:"inactive_anon,omitempty" yaml:"inactive_anon,omitempty"`
+			Activefile      int64 `json:"active_file,omitempty" yaml:"active_file,omitempty"`
+			Pgfault         int64 `json:"pgfault,omitempty" yaml:"pgfault,omitempty"`
+			InactiveFile    int64 `json:"inactive_file,omitempty" yaml:"inactive_file,omitempty"`
+			TotalPgpgin     int64 `json:"total_pgpgin,omitempty" yaml:"total_pgpgin,omitempty"`
+		} `json:"stats,omitempty" yaml:"stats,omitempty"`
+		MaxUsage int64 `json:"max_usage,omitempty" yaml:"max_usage,omitempty"`
+		Usage    int64 `json:"usage,omitempty" yaml:"usage,omitempty"`
+		Failcnt  int64 `json:"failcnt,omitempty" yaml:"failcnt,omitempty"`
+		Limit    int64 `json:"limit,omitempty" yaml:"limit,omitempty"`
+	} `json:"memory_stats,omitempty" yaml:"memory_stats,omitempty"`
+	// TODO(pedge): this is in the docker docs, but no data
+	//BlkioStats string `json:"blkio_stats,omitempty" yaml:"blkio_stats,omitempty"`
+	CPUStats struct {
+		CPUUsage struct {
+			PercpuUsage       []int64 `json:"percpu_usage,omitempty" yaml:"percpu_usage,omitempty"`
+			UsageInUsermode   int64   `json:"usage_in_usermode,omitempty" yaml:"usage_in_usermode,omitempty"`
+			TotalUsage        int64   `json:"total_usage,omitempty" yaml:"total_usage,omitempty"`
+			UsageInKernelmode int64   `json:"usage_in_kernelmode,omitempty" yaml:"usage_in_kernelmode,omitempty"`
+		} `json:"cpu_usage,omitempty" yaml:"cpu_usage,omitempty"`
+		SystemCPUUsage int64 `json:"system_cpu_usage,omitempty" yaml:"system_cpu_usage,omitempty"`
+		// TODO(pedge): this is in the docker docs, but no data
+		//ThrottlingData string `json:"throttling_data,omitempty" yaml:"throttling_data,omitempty"`
+	} `json:"cpu_stats,omitempty" yaml:"cpu_stats,omitempty"`
+}
+
+// Stats sends container statistics for the given container to the given channel.
+//
+// This function is blocking, similar to a streaming call for logs, and should be run
+// on a separate goroutine from the caller. Note that this function will block until
+// the given container is removed, not just exited. When finished, this function
+// will close the given channel.
+//
+// See http://goo.gl/DFMiYD for more details.
+func (c *Client) Stats(id string, statsC chan<- *Stats) (retErr error) {
```

Comment:
I would prefer to use a struct with options here, just in case Docker extends the API, so we can add new parameters to the method without breaking existing code. Something like:

```
type StatusOptions struct {
	Container string
	Stats     chan<- *Stats
}
```

At the moment, you won't use to build any querystring, but in the future we might need it.